agentnet/utils/layers.py

```diff
@@ -44,89 +47,135 @@ def get_layer_dtype(layer, default=None):
     return layer.output_dtype if hasattr(layer, "output_dtype") else default or theano.config.floatX
 
 
-# TODO (arogozhnikov) use names for outputs. Recommended interface
-#  tuple_layer = TupleLayer(dict(old_memory=old_memory, observations=observations))
-#  new_memory, action = tuple_layer['new_memory', 'action']
 
-class TupleLayer(MergeLayer):
+class DictLayer(MergeLayer):
```

Comment:
Imho dictlayer may lead to smoother conversion if anyone ever happens to implement him inside Lasagne.
If you prefer any of those you suggested, just point that out.

Humble opinion:
* Named Layer - introduces confusion since every layer has a name
* GraphLayer or DAGLayer or DictLayer - idk which is better.